File: 2016/luke05.hs
import Data.Maybe
import Data.List.Split

main :: IO ()
main = do content <- lines <$> readFile "luke05_data.txt"
          let decs = toDec <$> (splitOn ", " . init . tail . head $ content)
              half = take $ length decs `div` 2
          print $ toEng <$> zipWith (+) (half decs) (half $ reverse decs)

toDec :: String -> Integer
toDec n = fromMaybe 0 (lookup n (zip ["I", "II", "III", "IV", "V", "VI", "VII", "VIII", "IX", "X", "XI", "XII", "XIII"] [1..13]))

toEng :: Integer -> Char
toEng n = fromJust (lookup n (zip [1..26] ['a'..'z']))

File: 2016/luke05.rb
to_dec = Hash[%w(0 I II III IV V VI VII VIII IX X XI XII XIII).zip((0..13))]
to_eng = Hash[(1..26).zip(('a'..'z'))]

decs = File.readlines('luke05_data.txt').first
  .strip[1..-2].tr(',', '').split(/\s+/)
  .map { |n| to_dec[n] }

p decs[0...decs.size/2]
  .zip(decs.reverse[0...decs.size/2])
  .map { |n| to_eng[n.inject(:+)] }.join
